src/utils/TaggedCollections.fs

```diff
@@ -711,7 +711,7 @@ namespace Internal.Utilities.Collections.Tagged
 #if ONE 
         | MapOne of 'Key * 'T
 #endif
-        // Note: performance rumour has it that the data held in this node should be
+        // NOTE: performance rumour has it that the data held in this node should be
```

Comment:
Just remove the comment.  We could do work on immutable collections perf, but I don't think we should start with a random comment I wrote 8 years ago based on a corridor conversation :)  Systems.Collections.Immutable and the Roslyn immutable collections would be the place.